js/pages/myToDoList.js

```diff
@@ -728,8 +728,25 @@ const setModuleAttributes = (data, modules, collections) => {
         modules['Covid-19'].enabled = true;
     }
 
-    if(data['289750687'] === 353358909) {
+    if(data[fieldMapping.menstrualSurveyEligible] === 353358909) {
         modules['Menstrual Cycle'].enabled = true;
+
+        if(data[fieldMapping.MenstrualCycle.statusFlag] != fieldMapping.moduleStatus.submitted) {
+
+            const currentDate = Date.now();
+            const milllisecondsToDays = 1000 * 60 * 60 * 24;
+
+            if(data[fieldMapping.Biospecimen.statusFlag] === fieldMapping.moduleStatus.submitted) {
+                if((currentDate - new Date(data[fieldMapping.Biospecimen.completeTs]).getTime()) / milllisecondsToDays > 45) {
+                    modules['Menstrual Cycle'].enabled = false;
+                }
+            }
```

Comment:
For the date check, calculation `(currentTime - savedTime)/millisecondsPerDay > 45` can be reformulated to `savedTime < currentTime - 45Days`, and then further to `savedTime < cutoffTime` with 'cutoffTime' refers to 'currentTime - 45Days'.
With above info, we may be able to make date calculation faster:
```js
let cutoffTime = new Date();
cutoffTime.setDate(cutoffTime.getDate() -45);
if (data[fieldMapping.Biospecimen.statusFlag] === fieldMapping.moduleStatus.submitted) {
  if (data[fieldMapping.Biospecimen.completeTs] < cutoffTime) {
                    modules['Menstrual Cycle'].enabled = false;
                }
            }
```

Reply:
I like it! Will implement the changes, thanks Warren :)